great_expectations/core/usage_statistics/anonymizers/profiler_run_anonymizer.py

```diff
@@ -0,0 +1,197 @@
+import logging
+from numbers import Number
+from typing import Any, Dict, List, Optional, Union
+
+from great_expectations.core.usage_statistics.anonymizers.anonymizer import Anonymizer
+from great_expectations.core.usage_statistics.anonymizers.batch_request_anonymizer import (
+    BatchRequestAnonymizer,
+)
+from great_expectations.util import deep_filter_properties_iterable
+
+logger = logging.getLogger(__name__)
+
+
+class ProfilerRunAnonymizer(Anonymizer):
+    def __init__(self, salt: Optional[str] = None) -> None:
+        super().__init__(salt=salt)
+
+        self._salt = salt
+        self._batch_request_anonymizer = BatchRequestAnonymizer(self._salt)
+
+    def anonymize_profiler_run(self, *args: List[Any], **kwargs: dict) -> dict:
+        """
+        Traverse the entire RuleBasedProfiler configuration structure (as per its formal, validated Marshmallow schema) and
+        anonymize every field that can be customized by a user (public fields are recorded as their original names).
+        """
+
+        name: Optional[str] = kwargs.get("name")
+        anonymized_name: Optional[str] = self.anonymize(name)
+
+        config_version: Union[Number, str] = kwargs.get("config_version", 1.0)
+        variable_count: Number = kwargs.get("variable_count", 0)
```

Comment:
I thought the choice for `Number` type here was super-interesting. What was the reasoning behind it?

Reply:
that's a great question! I stole this bit of code from the `CheckpointRunAnonymizer` 😬 

I think float makes more sense!